dataflow/src/org/checkerframework/dataflow/analysis/Store.java

```diff
@@ -54,6 +54,25 @@
     S leastUpperBound(S other);
 
     /**
+     * Compute an upper bound of two stores that is wider than the least upper bound of the two
+     * stores.
+     *
+     * <p>(This method is optional, if not required, implementations should call leastUpperBound.)
+     *
+     * <p><em>Important</em>: This method must fulfill the following contract:
+     *
+     * <ul>
+     *   <li>Does not change {@code this}.
+     *   <li>Does not change {@code other}.
+     *   <li>Returns a fresh object which is not aliased yet.
+     *   <li>Returns an object of the same (dynamic) type as {@code this}, even if the signature is
```

Comment:
````
class MyStore implements A {
  MyStore lub(MyStore s) {...}
}

class MySubStore extends MyStore{
  // Doesn't implement lub
}
````

If lub is called on an object with type MySubStore, the result should have the dynamic type of MySubStore.